[COLAB-2344] Fix feeds dump export button

jobachhu opened the pull request:
This PR fixes two things:
1. the export button in the admin tab does currently not include the activity body
2. the export button in the activity feed uses a completely separate CsvExport that does not work

After fixing 1, this PR simply uses the working CsvConverter from 1 to export the data in 2 (less duplication - yay!).

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cci-mit/xcolab/49)
<!-- Reviewable:end -->

carlosbpf commented:
Reviewed 12 of 12 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[view/src/main/java/org/xcolab/view/pages/contestmanagement/utils/ActivityCsvWriter.java, line 26 at r1](https://reviewable.io:443/reviews/cci-mit/xcolab/49#-KwblaQWxfNV4Db5uzcw:-KwblaQWxfNV4Db5uzcx:b86ax8j) ([raw file](https://github.com/cci-mit/xcolab/blob/bea3f4b6d31fd34fc3d244a767f5a145e862c72b/view/src/main/java/org/xcolab/view/pages/contestmanagement/utils/ActivityCsvWriter.java#L26)):*
> ```Java
> public class ActivityCsvWriter extends CsvResponseWriter {
> 
>     private static final Logger log = LoggerFactory.getLogger(ActivityCsvWriter.class);
> ```

We often use the _log, just to keep things consistent.

---

*[view/src/main/java/org/xcolab/view/pages/contestmanagement/utils/ActivityCsvWriter.java, line 61 at r1](https://reviewable.io:443/reviews/cci-mit/xcolab/49#-Kwbp0bUUXaTGcDEzTBG:-Kwbp0bVMPxAe2pPlyqI:b-s2qigz) ([raw file](https://github.com/cci-mit/xcolab/blob/bea3f4b6d31fd34fc3d244a767f5a145e862c72b/view/src/main/java/org/xcolab/view/pages/contestmanagement/utils/ActivityCsvWriter.java#L61)):*
> ```Java
> 
>             List<String> row = new ArrayList<>();
>             addValue(row, member != null ? member.getId_() : "Member not found");
> ```

Wouldn't this strings be better suited in a resource file? Maybe not so complex solution would be to extract them into static String constants.

---

*[view/src/main/java/org/xcolab/view/util/CsvResponseWriter.java, line 105 at r1](https://reviewable.io:443/reviews/cci-mit/xcolab/49#-KwbpZ5aMcXbsbSKDNU3:-KwbpZ5aMcXbsbSKDNU4:b-jko798) ([raw file](https://github.com/cci-mit/xcolab/blob/bea3f4b6d31fd34fc3d244a767f5a145e862c72b/view/src/main/java/org/xcolab/view/util/CsvResponseWriter.java#L105)):*
> ```Java
>         csvWriter.close();
>     }
> }
> ```

Really neat solution, abstracting the CSV download

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cci-mit/xcolab/49)*
<!-- Sent from Reviewable.io -->

jobachhu commented:
Review status: 9 of 10 files reviewed at latest revision, 3 unresolved discussions.

---

*[view/src/main/java/org/xcolab/view/pages/contestmanagement/utils/ActivityCsvWriter.java, line 26 at r1](https://reviewable.io:443/reviews/cci-mit/xcolab/49#-KwblaQWxfNV4Db5uzcw:-KwbuXxvDEcobkDQFH8w:b25x716) ([raw file](https://github.com/cci-mit/xcolab/blob/bea3f4b6d31fd34fc3d244a767f5a145e862c72b/view/src/main/java/org/xcolab/view/pages/contestmanagement/utils/ActivityCsvWriter.java#L26)):*
<details><summary><i>Previously, carlosbpf (Carlos Paula) wrote…</i></summary><blockquote>

We often use the _log, just to keep things consistent.
</blockquote></details>

Good point, I changed it to `_log`.

---

*[view/src/main/java/org/xcolab/view/pages/contestmanagement/utils/ActivityCsvWriter.java, line 61 at r1](https://reviewable.io:443/reviews/cci-mit/xcolab/49#-Kwbp0bUUXaTGcDEzTBG:-Kwbw4-v6RA92acmTiy4:b-uty96k) ([raw file](https://github.com/cci-mit/xcolab/blob/bea3f4b6d31fd34fc3d244a767f5a145e862c72b/view/src/main/java/org/xcolab/view/pages/contestmanagement/utils/ActivityCsvWriter.java#L61)):*
<details><summary><i>Previously, carlosbpf (Carlos Paula) wrote…</i></summary><blockquote>

Wouldn't this strings be better suited in a resource file? Maybe not so complex solution would be to extract them into static String constants.
</blockquote></details>

You're right - the String shouldn't be hard-coded and repeated. I extracted a constant (it's an admin function, so we don't need to internationalize it).

---

*[view/src/main/java/org/xcolab/view/util/CsvResponseWriter.java, line 105 at r1](https://reviewable.io:443/reviews/cci-mit/xcolab/49#-KwbpZ5aMcXbsbSKDNU3:-KwbwGHRwEsPq3mFbm7B:b-ucfknh) ([raw file](https://github.com/cci-mit/xcolab/blob/bea3f4b6d31fd34fc3d244a767f5a145e862c72b/view/src/main/java/org/xcolab/view/util/CsvResponseWriter.java#L105)):*
<details><summary><i>Previously, carlosbpf (Carlos Paula) wrote…</i></summary><blockquote>

Really neat solution, abstracting the CSV download
</blockquote></details>

Thanks!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cci-mit/xcolab/49)*
<!-- Sent from Reviewable.io -->

carlosbpf commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cci-mit/xcolab/49)*
<!-- Sent from Reviewable.io -->
